ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/GamePanel.java

```diff
@@ -251,73 +250,121 @@ private void initComponents() {
-        constraints.weighty = 0.0;
-        constraints.ipadx = 0;
-
-        // add the Black Team Label
-        constraints.anchor = GridBagConstraints.BASELINE;
-        constraints.gridy = 0;
-        constraints.insets = new Insets(10, 0, 10, 10);
-        jailAndLabelPanel.add(mTeamLabels.get(1), constraints);
-
-        // add the Black timer
-        constraints.gridy = 1;
-        constraints.insets = new Insets(0, 0, 10, 10);
-        // TODO: This currently assumes a two-person game.
-//		if (!ChessTimer.isNoTimer(mTimers[1]))
-//			jailAndLabelPanel.add(new ChessTimerLabel(mTimers[1]), constraints);
-
-        // add the Black Jail
-        constraints.gridy = 2;
-        constraints.weighty = 0.7;
-        jailAndLabelPanel.add(mJails[1], constraints);
-
-        // adds the undo button
-        constraints.weighty = 0.0;
-        constraints.gridy = 3;
-        jailAndLabelPanel.add(undoButton, constraints);
-
-        // adds the White Jail
-        constraints.gridy = 4;
-        constraints.weighty = 0.7;
-        jailAndLabelPanel.add(mJails[0], constraints);
-
-        // adds the White timer
-        constraints.gridy = 5;
-        // TODO: This assumes a two-player game
-//		jailAndLabelPanel.add(new ChessTimerLabel(mTimers[0]), constraints);
-
-        //Add the White Team Label
-        constraints.gridy = 6;
-        constraints.weighty = 0.275;
-        constraints.insets = new Insets(0, 0, 10, 10);
-        jailAndLabelPanel.add(mTeamLabels.get(0), constraints);
+        createTeamPanels();
 
         constraints.gridx = (boards.length * 2);
-        constraints.gridy = 0;
-        constraints.weighty = 0.0;
         constraints.weightx = 0.3;
         constraints.insets = new Insets(0, 25, 0, 10);
-        add(jailAndLabelPanel, constraints);
+        add(mTabbedPane, constraints);
 
-//		resetTimers();
         boardRefresh();
         refreshStatus();
     }
 
+    private void createTeamPanels(){
+        for(int team = 0; team < mTeamLabels.size(); team++) {
+            GridBagConstraints constraints = new GridBagConstraints();
+            JButton undoButton = new JButton(Messages.getString("PlayGamePanel.undo"));
+            undoButton.addActionListener(event -> {
+                mGame.undoMove();
+                refreshStatus();
+                boardRefresh();
+            });
+            JButton forward = new JButton("->");
+            JButton back = new JButton("<-");
+
+            JPanel jailAndLabelPanel = new JPanel();
+            jailAndLabelPanel.setLayout(new GridBagLayout());
+            jailAndLabelPanel.setOpaque(false);
+
+            JPanel teamMetaData = new JPanel();
+            teamMetaData.setOpaque(false);
+            teamMetaData.setLayout(new GridBagLayout());
+            teamMetaData.setBorder(BorderFactory.createTitledBorder(Messages.getString("GamePanel.info")));
+
+            GridBagConstraints teamConstraint = new GridBagConstraints();
+
+            //Add Team name label
```

Comment:
Needs a space after the `//` and the comment should start with a lowercase letter: `// add team name label`

Same for other comments